vars/githubPublishRelease.groovy

```diff
@@ -1,198 +1,42 @@
-import com.sap.piper.JsonUtils
-
-import static com.sap.piper.Prerequisites.checkScript
-
-import com.sap.piper.GenerateDocumentation
+import com.sap.piper.PiperGoUtils
 import com.sap.piper.Utils
-import com.sap.piper.ConfigurationHelper
-
-import groovy.text.GStringTemplateEngine
 import groovy.transform.Field
 
-@Field String STEP_NAME = getClass().getName()
-
-@Field Set GENERAL_CONFIG_KEYS = [
-    /** Allows to overwrite the GitHub API url.*/
-    'githubApiUrl',
-    /**
-     * Allows to overwrite the GitHub token credentials id.
-     * @possibleValues Jenkins credential id
-     */
-    'githubTokenCredentialsId',
-    /** Allows to overwrite the GitHub url.*/
-    'githubServerUrl'
-]
+import static com.sap.piper.Prerequisites.checkScript
 
-@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
-    /**
-     * If it is set to `true`, a list of all closed issues and merged pull-requests since the last release will added below the `releaseBodyHeader`.
-     * @possibleValues `true`, `false`
-     */
-    'addClosedIssues',
-    /**
-     * If you set `addDeltaToLastRelease` to `true`, a link will be added to the relese information that brings up all commits since the last release.
-     * @possibleValues `true`, `false`
-     */
-    'addDeltaToLastRelease',
-    /** Allows to pass additional filter criteria for retrieving closed issues since the last release. Additional criteria could be for example specific `label`, or `filter` according to [GitHub API documentation](https://developer.github.com/v3/issues/).*/
-    'customFilterExtension',
-    /** Allows to exclude issues with dedicated labels. Usage is like `excludeLabels: ['label1', 'label2']`.*/
-    'excludeLabels',
-    /** Allows to overwrite the GitHub organitation.*/
-    'githubOrg',
-    /** Allows to overwrite the GitHub repository.*/
-    'githubRepo',
-    /** Allows to specify the content which will appear for the release.
-     * It is possible to define it as Groovy template as well in order to bring in dynamic information.
-     * Following information can be used: everything contained in `config` as well as information from `commonPipelineEnvironment`.
-     */
-    'releaseBodyHeader',
-    /** Defines the version number which will be written as tag as well as release name.*/
-    'version'
-])
+@Field String STEP_NAME = getClass().getName()
+@Field String METADATA_FILE = 'metadata/githubrelease.yaml'
 
-@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+//Metadata maintained in file resources/metadata/githubrelease.yaml
```

Comment:
Add prefix to make this clickable in VSCode with [project-links](https://marketplace.visualstudio.com/items?itemName=KyleDavidE.vscode-project-links) extension.
```suggestion
//Metadata maintained in file project://resources/metadata/githubrelease.yaml
```